test/test_other.py

```diff
@@ -10268,6 +10268,51 @@ def test_wasm_features_section(self, args):
     self.run_process([EMCC, test_file('hello_world.c'), '-O2'] + args)
     self.verify_custom_sec_existence('a.out.wasm', 'target_features', False)
 
+  def test_wasm_features(self):
+    # Test that wasm features are explicitly enabled or disabled based on target engine version
+    def verify_features_sec(feature, expect_in, linked=False):
+      with webassembly.Module('a.out.wasm' if linked else 'hello_world.o') as module:
+        features = module.get_target_features()
+      if expect_in:
+        self.assertTrue(feature in features and
+                        features[feature] == webassembly.TargetFeaturePrefix.USED,
+                        f'{feature} missing from wasm file')
+      else:
+        self.assertFalse(feature in features,
+                         f'{feature} unexpectedly found in wasm file')
+
+    def verify_features_sec_linked(feature, expect_in):
+      return verify_features_sec(feature, expect_in, linked=True)
+
+    def compile(flags):
+      self.run_process([EMCC, test_file('hello_world.c')] + flags)
+
+    compile(['-c'])
+    verify_features_sec('bulk-memory', False)
+    verify_features_sec('nontrapping-fptoint', False)
+    verify_features_sec('sign-ext', True)
+    verify_features_sec('mutable-globals', True)
+    verify_features_sec('multivalue', True)
+    verify_features_sec('reference-types', True)
+
+    compile(['-mnontrapping-fptoint', '-c'])
+    verify_features_sec('nontrapping-fptoint', True)
+
+    # BIGINT causes binaryen to not run, and keeps the target_features section after link
+    # Setting this SAFARI_VERSION should enable bulk memory because it links in emscripten_memcpy_bulkmem
+    # However it does not enable nontrapping-fptoint yet because it has no effect at compile time and
+    # no libraries include nontrapping yet.
+    compile(['-sMIN_SAFARI_VERSION=150000', '-sWASM_BIGINT'])
+    verify_features_sec_linked('sign-ext', True)
+    verify_features_sec_linked('mutable-globals', True)
+    verify_features_sec_linked('multivalue', True)
+    verify_features_sec_linked('bulk-memory', True)
+    verify_features_sec_linked('nontrapping-fptoint', False)
+
+    compile(['-sMIN_SAFARI_VERSION=150000', '-mno-bulk-memory', '-sWASM_BIGINT'])
+    # FIXME? -mno-bulk-memory at link time does not override MIN_SAFARI_VERSION. it probably should?
```

Comment:
one weirdness we have currently is that the -m flags (including -mno-signext currently) do not work at link time to disable features (and e.g. do not trigger the lowering passes). Only the the feature_matrix/browser version code does this. So there's currently no way to specify the feature behavior at link time on a granular level. I'm not sure what exactly the use case for this would be, but it wouldn't surprise me too much if there were one. Maybe we should think about that.